src/code_80718D8.c

```diff
@@ -0,0 +1,168 @@
+extern s16 gUnknown_80F4DAC;
+extern u8 *gUnknown_80FED68[];
+
+extern void SetMessageArgument(char[], struct Entity*, u32);
+extern void sub_806B7F8(struct unkStruct_806B7F8 *, u32);
+extern void sub_805239C(struct Entity *r0, const char r1[]);
+extern void sub_8042920(struct Entity *r0);
+extern s16 sub_803D970(u32);
+extern s32 sub_803DA20(s32 param_1);
+extern bool8 sub_806AA0C(s32, u32);
+extern bool8 sub_8083660(struct Position *param_1);
+
+void sub_8071B48(void)
+{
+  struct Tile *tile;
+  struct Entity *entity2;
+  struct Entity *entity;
+  int index;
+  int index1;
+  int counter;
+  struct Entity *entityPtr;
+  s32 level;
+  struct Dungeon *dungeon; // feels like a hack
+  struct unkStruct_806B7F8 local_2c;
+  
+  entityPtr = NULL;
+  dungeon = gDungeon;
+  if ((dungeon->unk664 != 0) &&
+     ((dungeon->dungeonLocation.id != DUNGEON_METEOR_CAVE || (dungeon->fill37FD == 0)))) {
+    dungeon->unk662++;
+    if (dungeon->unk66E == 0) {
+      if ( dungeon->unk662 < gUnknown_80F4DAA) {
+        return;
+      }
+    }
+    else {
+      if ( dungeon->unk662 < gUnknown_80F4DAC) {
+        return;
+      }
+    }
+
+    if (dungeon->unk662 > 900) {
+      entity2 = dungeon->unk17B34;
+      if ((EntityExists(entity2)) && (entity2->spawnGenID == dungeon->unk17B40)) {
+        SetMessageArgument(gAvailablePokemonNames,entity2,0);
+        entityPtr = entity2;
+      }
+    }
+      
+      dungeon->unk662 = 0;
+      dungeon->unk17B34 = NULL;
+        
+      counter = 0;
+      for(index = 0; index < DUNGEON_MAX_WILD_POKEMON; index++)
+      {
+        entity = dungeon->wildPokemon[index];
+        if ((EntityExists(entity)) &&
+           (tile = GetTile((entity->pos).x, (entity->pos).y),
+           (tile->terrainType & 0x100) == 0)) {
+          counter++;
+        }
+      }
+      if (dungeon->unk66E == 0) {
+        if (gDungeon->unk3A0C == 0xff) {
+            if (counter >= gUnknown_80F4DA6) {
+              return;
+            }
+        }
+        else
+        {         
+            if (counter >= gUnknown_80F4DA8) {
+              return;
+            }
+        }
+
+      }
+      if (entityPtr != NULL) {
+        sub_805239C(0,*gUnknown_80FED68);
+        sub_8042920(entityPtr);
+      }
+      if (dungeon->unk66E != 0) {
+        index1 = 0x17c;
+      }
+      else {
+        index1 = sub_803D970(0);
+      }
+
+      level = sub_803DA20(index1);
+      if (level == 0) {
+        level = 1;
+      }
+      if ((sub_806AA0C(index1, 0)) && (sub_8083660(&local_2c.pos))) {
+        local_2c.species = index1;
+        local_2c.level = level;
+        local_2c.unk2 = 0;
+        if(DungeonRandInt(100) < GetRandomMovementChance(gDungeon->dungeonLocation.id))
+        {
+            local_2c.unk4 = 1;
+        }
+        else
+        {
+            local_2c.unk4 = 0;
+        }
+        local_2c.unk10 = 0;
+        sub_806B7F8(&local_2c, 0);
+      }
+  }
+}
+
+static inline bool8 sub_8071D4C_sub(EntityInfo *pokemonInfo)
```

Comment:
This function is called `JoinLocationCannotUseItems` in other places where it is duplicated. We can continue to suffix numbers onto it rather than leaving it unlabeled.

Reply:
I can just use `JoinLocationCannotUseItems` since it's inline. It won't conflict with the others.